tests/integ/compiler/test_query_generator.py

```diff
@@ -99,13 +99,13 @@ def check_generated_plan_queries(plan: SnowflakePlan) -> None:
 
 
 def verify_multiple_create_queries(
-    plan_queries: List[str], post_action_queries: List[str]
+    plan_queries: List[str], post_action_queries: List[str], num_queries: int
 ) -> None:
-    assert len(plan_queries) == 3
+    assert len(plan_queries) == num_queries
     assert plan_queries[0].startswith("CREATE")
     assert plan_queries[1].startswith("INSERT")
-    assert plan_queries[2].startswith("SELECT")
-    assert len(post_action_queries) == 1
+    assert plan_queries[-1].startswith("SELECT")
+    assert len(post_action_queries) == int(num_queries / 2)
```

Comment:
`len(post_action_queries) == int(num_queries / 2)` this hold in most cases but might not be true in general. We can add a comment mentioning that.

Reply:
good point, when create is not followed with INSERT, this will not be true. Added comment for this test utitlity function